test/integration/puppetlabs/trapperkeeper/services/scheduler/scheduler_service_test.clj

```diff
@@ -178,6 +179,18 @@
           (testing "throws Illegal Argument exception for invalid cron string"
             (is (thrown? IllegalArgumentException
                          (cron service "**invalid**" job :some-group-identifier)))))))))
+(defn add-seconds [date seconds]
+  (let [millis (+ (.getTime date) (* seconds 1000))]
+    (java.util.Date. millis)))
+
+(deftest ^:integration test-cron-next-valid-time
+  (testing "cron-next-valid-time returns the next runtime of a cron specification from a given date"
+    (with-app-with-empty-config app [scheduler-service]
+      (let [cron-string "0/10 * * * * ?" ; every 10 seconds
+            current-time (Date.)
+            expected-execution-time (add-seconds current-time 10)
+            next-execution-time (cron-next-valid-time cron-string current-time)]
```

Comment:
You need a service instance to pass into the function as the first argument, similar to 
```
(let [job-id (cron service cron-string job :some-group-identifier)]
```

Reply:
So something like:
```suggestion
            service (tk/get-service app :SchedulerService)
            next-execution-time (cron-next-valid-time service cron-string current-time)]
```